src/Compiler/Microsoft.AspNetCore.Razor.Language/test/IntegrationTests/ComponentCodeGenerationTestBase.cs

```diff
@@ -2501,10 +2501,7 @@ public class MyComponent : ComponentBase
         // Assert
         AssertDocumentNodeMatchesBaseline(generated.CodeDocument);
         AssertCSharpDocumentMatchesBaseline(generated.CodeDocument);
-        CompileToAssembly(generated,
-            // x:\dir\subdir\Test\TestComponent.cshtml(1,61): error CS1503: Argument 2: cannot convert from 'Microsoft.AspNetCore.Components.EventCallback<int>' to 'System.Action<int>'
-            //                                                             UpdateValue
-            Diagnostic(ErrorCode.ERR_BadArgType, "UpdateValue").WithArguments("2", "Microsoft.AspNetCore.Components.EventCallback<int>", "System.Action<int>").WithLocation(1, 61));
+        CompileToAssembly(generated);
```

Comment:
Can u help me understand why this was failing before? Is it just our old ref assembly approach was incomplete?

Reply:
We did not use ref assemblies. We used "shims". In this specific case, the error was reported because we are missing this overload that is present in the ref assemblies:

https://github.com/dotnet/aspnetcore/blob/daf411cf171c5de0c9a31bf9813a4c63063a964b/src/Components/Components/src/CompilerServices/RuntimeHelpers.cs#L68

In our shims we have only these two overloads:

https://github.com/dotnet/razor/blob/62c5f524fb221efc1bcff04c03352a8ac705cd7b/src/Shared/Microsoft.AspNetCore.Razor.Test.ComponentShim/Microsoft.AspNetCore.Components.netstandard2.0.cs#L426-L427